actions/microsoft-onedrive/microsoft_onedrive/onedrive_get_action.py

```diff
@@ -0,0 +1,331 @@
+    token: OAuth2Secret[
+        Literal["microsoft"],
+        List[Literal["Files.Read"]],
+    ],
+    params: GetOneDriveItemByIdParams,
+) -> Response[dict]:
+    """
+    Retrieve detailed information about a file or folder in OneDrive using its ID.
+    Useful also to find parent folders.
+
+    Args:
+        token: OAuth2 token with Files.Read permission.
+        params: Pydantic model containing the item_id parameter.
+
+    Returns:
+        Dictionary with detailed information about the requested item.
+    """
+    headers = build_headers(token)
+    url = f"{BASE_GRAPH_URL}/me/drive/items/{params.item_id}"
+
+    try:
+        response = requests.get(url, headers=headers)
+        response.raise_for_status()
+
+        item_data = response.json()
+
+        result = {
+            "id": item_data.get("id"),
+            "name": item_data.get("name"),
+            "type": "folder" if "folder" in item_data else "file",
+            "size": item_data.get("size"),
+            "created_datetime": item_data.get("createdDateTime"),
+            "last_modified_datetime": item_data.get("lastModifiedDateTime"),
+            "web_url": item_data.get("webUrl"),
+        }
+
+        if "folder" in item_data:
+            result["child_count"] = item_data["folder"].get("childCount")
+        elif "file" in item_data:
+            result["mime_type"] = item_data["file"].get("mimeType")
+
+        if "parentReference" in item_data:
+            result["parent_folder"] = {
+                "id": item_data["parentReference"].get("id"),
+                "path": item_data["parentReference"].get("path"),
+            }
+
+        return Response(result=result)
+
+    except requests.HTTPError as e:
+        status_code = e.response.status_code
+        if status_code == 400:
+            raise ActionError(
+                f"Bad request. The item ID '{params.item_id}' might be invalid."
+            )
+        elif status_code == 401:
+            raise ActionError("Unauthorized. Please check your authentication token.")
+        elif status_code == 403:
+            raise ActionError(
+                "Access forbidden. You may not have permission to access this item."
+            )
+        elif status_code == 404:
+            raise ActionError(f"Item not found: {params.item_id}")
+        else:
+            raise ActionError(f"Failed to retrieve OneDrive item: HTTP {status_code}")
+    except requests.RequestException as e:
+        raise ActionError(f"Network error while retrieving OneDrive item: {str(e)}")
+    except Exception as e:
+        raise ActionError(f"Unexpected error while retrieving OneDrive item: {str(e)}")
+
+
+@action
+def list_all_onedrive_folders_recursively(
+    token: OAuth2Secret[
+        Literal["microsoft"],
+        List[Literal["Files.Read"]],
+    ],
+    root_folder: str = "/",
+) -> Response[dict]:
+    """
+    Recursively list all folders in OneDrive starting from a specified root folder.
```

Comment:
We follow Google style docstrings therefore the docstring should start from the first line.